### DISCUSSION TITLE
Update with new data

### DISCUSSION THREAD

**Discussion by caxschoen** [imartinez/privateGPT#476] [General]
Is there any function available to update with furher data? Or does it need to read all data again?

**Comment by johnbrisbin** [comment]
The current version handles this reasonably well. (Always backup your data, just sayin')
Just add any new items and launch the ingest task. The code scans the existing database and collects the names of all the source documents, then uses that as an exclusion list for further processing. In other words, pre-existing documents are ignored on subsequent passes.
Just run ingest again. (After you backup the db folder).
